cisco_vxlan_global --> cisco_overlay_global, no longer ensurable

glennmatthews opened the pull request:
Counterpart to cisco/cisco-network-node-utils#220.

* Rename 'cisco_vxlan_global' to 'cisco_overlay_global'
* Because it doesn't make any sense for the global config to be 'absent' (some of the properties ALWAYS have default values), this type is no longer `ensurable`. 
* Beaker test passes against both I2 and I3 images on N9k:

```
Begin tests/beaker_tests/cisco_overlay_global/test_overlay_global.rb

TestCase :: Resource cisco_overlay_global
  
  ------------------------------------------------------------
  Section 1. Default Property Testing
  
  * TestStep :: 1.1 Default Properties :: MANIFEST    
  1.1 Default Properties :: MANIFEST     :: PASS
  
  * TestStep :: 1.1 Default Properties :: RESOURCE    
    1.1 Default Properties :: RESOURCE     :: PASS
  
  * TestStep :: 1.1 Default Properties :: IDEMPOTENCE 
    1.1 Default Properties :: IDEMPOTENCE  :: PASS
  
  ------------------------------------------------------------
  Section 2. Non Default Property Testing
  
  * TestStep :: 2.1 Non Default Properties :: MANIFEST    
  2.1 Non Default Properties :: MANIFEST     :: PASS
  
  * TestStep :: 2.1 Non Default Properties :: RESOURCE    
    2.1 Non Default Properties :: RESOURCE     :: PASS
  
  * TestStep :: 2.1 Non Default Properties :: IDEMPOTENCE 
    2.1 Non Default Properties :: IDEMPOTENCE  :: PASS
TestCase :: Resource cisco_overlay_global :: End
tests/beaker_tests/cisco_overlay_global/test_overlay_global.rb passed in 42.20 seconds
      Test Suite: tests @ 2016-02-10 10:19:47 -0500

      - Host Configuration Summary -


              - Test Case Summary for suite 'tests' -
       Total Suite Time: 42.20 seconds
      Average Test Time: 42.20 seconds
              Attempted: 1
                 Passed: 1
                 Failed: 0
                Errored: 0
                Skipped: 0
                Pending: 0
                  Total: 1
```

chrisvanheuveln commented on lib/puppet/provider/cisco_overlay_global/nxapi.rb:
```diff
@@ -41,60 +41,44 @@
 
-  PuppetX::Cisco::AutoGen.mk_puppet_methods(:non_bool, self, '@vxlan_global',
-                                            VXLAN_GLOBAL_PROPS)
+  PuppetX::Cisco::AutoGen.mk_puppet_methods(:non_bool, self, '@overlay_global',
```
Just an fyi - Mike and I have been using `@nu` for this var name on the new properties. It makes it a little more clear which object it's referring to and also allows us to cut & paste the code with fewer changes.

glennmatthews replied:
I'll update this - thanks.

chrisvanheuveln commented on lib/puppet/provider/cisco_overlay_global/nxapi.rb:
```diff
@@ -41,60 +41,44 @@
-      ensure: :present,
-      name:   'default',
-    }
+  def self.properties_get(overlay_global)
```
In a similar vein, we're using `nu_obj` here. 

No need to change these now, just letting you know.

chrisvanheuveln commented:
:+1:

glennmatthews commented:
Addressed Chris's comments and added a preclean step back into the test.

```
Begin tests/beaker_tests/cisco_overlay_global/test_overlay_global.rb

TestCase :: Resource cisco_overlay_global
  
  no nv overlay evpn
  
  l2rib dup-host-mac-detection default
  
  * TestStep :: Preclean :: MANIFEST    
  Preclean :: MANIFEST     :: PASS
  
  * TestStep :: Preclean :: RESOURCE    
    Preclean :: RESOURCE     :: PASS
  
  * TestStep :: Preclean :: IDEMPOTENCE 
    Preclean :: IDEMPOTENCE  :: PASS
  
  ------------------------------------------------------------
  Section 1. Default Property Testing
  
  * TestStep :: 1.1 Default Properties :: MANIFEST    
  1.1 Default Properties :: MANIFEST     :: PASS
  
  * TestStep :: 1.1 Default Properties :: RESOURCE    
    1.1 Default Properties :: RESOURCE     :: PASS
  
  * TestStep :: 1.1 Default Properties :: IDEMPOTENCE 
    1.1 Default Properties :: IDEMPOTENCE  :: PASS
  
  ------------------------------------------------------------
  Section 2. Non Default Property Testing
  
  * TestStep :: 2.1 Non Default Properties :: MANIFEST    
  2.1 Non Default Properties :: MANIFEST     :: PASS
  
  * TestStep :: 2.1 Non Default Properties :: RESOURCE    
    2.1 Non Default Properties :: RESOURCE     :: PASS
  
  * TestStep :: 2.1 Non Default Properties :: IDEMPOTENCE 
    2.1 Non Default Properties :: IDEMPOTENCE  :: PASS
TestCase :: Resource cisco_overlay_global :: End
tests/beaker_tests/cisco_overlay_global/test_overlay_global.rb passed in 62.09 seconds
      Test Suite: tests @ 2016-02-10 11:01:59 -0500

      - Host Configuration Summary -


              - Test Case Summary for suite 'tests' -
       Total Suite Time: 62.09 seconds
      Average Test Time: 62.09 seconds
              Attempted: 1
                 Passed: 1
                 Failed: 0
                Errored: 0
                Skipped: 0
                Pending: 0
                  Total: 1
```

mikewiebe commented on tests/beaker_tests/cisco_overlay_global/test_overlay_global.rb:
```diff
@@ -103,8 +102,12 @@
+  },
 }
 
 tests['default_properties'] = {
```
Please add `platform` key `n9k` to the top-level test hash and add call to `platform_supports_test` api.

https://github.com/cisco/cisco-network-puppet-module/blob/release_1.2.0/tests/beaker_tests/cisco_vxlan_vtep/test_vxlan_vtep.rb#L75

https://github.com/cisco/cisco-network-puppet-module/blob/release_1.2.0/tests/beaker_tests/cisco_vxlan_vtep/test_vxlan_vtep.rb#L196

glennmatthews replied:
Added. Also wrapped the command_config calls in preclean with this check as I suspect they will probably fail on other platforms.

mikewiebe commented:
:+1:
